Fails to build with musl libc, undefined reference to `backtrace'

### Godot version

4.0 alpha 1, commit 2b61d02f7ef5cd820020747f55107c6339324495

### System information

Gentoo, with musl libc overlay

### Issue description

When linking, it does not find `backtrace` or `backtrace_symbols`. Unlike glibc, the musl libc does not come with those symbols.

From SCons output:
```
[100%] Linking Static Library core/libcore.linuxbsd.tools.64.a ...
Ranlib Library core/libcore.linuxbsd.tools.64.a ...
[100%] Linking Program bin/godot.linuxbsd.tools.64 ...
/usr/lib/gcc/x86_64-gentoo-linux-musl/11.2.0/../../../../x86_64-gentoo-linux-musl/bin/ld: platform/linuxbsd/crash_handler_linuxbsd.linuxbsd.tools.64.o: in function `handle_crash(int)':
/home/user/Projects/godot/platform/linuxbsd/crash_handler_linuxbsd.cpp:56: undefined reference to `backtrace'
/usr/lib/gcc/x86_64-gentoo-linux-musl/11.2.0/../../../../x86_64-gentoo-linux-musl/bin/ld: /home/user/Projects/godot/platform/linuxbsd/crash_handler_linuxbsd.cpp:80: undefined reference to `backtrace_symbols'
collect2: error: ld returned 1 exit status
scons: *** [bin/godot.linuxbsd.tools.64] Error 1
scons: building terminated because of errors.
[Time elapsed: 00:04:45.680]
```

### Steps to reproduce

1. Have musl libc, and libexecinfo installed
2. Clone repo
3. Run scons

### Minimal reproduction project

_No response_

--- a/platform/linuxbsd/detect.py
+++ b/platform/linuxbsd/detect.py
@@ -363,8 +363,10 @@ def configure(env: "Environment"):
     if platform.system() == "Linux":
         env.Append(LIBS=["dl"])
 
-    if platform.system().find("BSD") >= 0:
-        env["execinfo"] = True
+    if not env["execinfo"] and platform.libc_ver()[0] != "glibc":
+        # The default crash handler depends on glibc, so if the host uses
+        # a different libc (BSD libc, musl), fall back to libexecinfo.
+        print("Note: Using `execinfo=yes` for the crash handler as required on platforms where glibc is missing.")
 
     if env["execinfo"]:
         env.Append(LIBS=["execinfo"])

